tests/mocks/mock_failing_prompt_driver.py

```diff
@@ -4,41 +4,22 @@
 
 from attrs import define
 
-from griptape.artifacts import TextArtifact
-from griptape.common import DeltaMessage, Message, PromptStack, TextDeltaMessageContent, TextMessageContent
 from griptape.drivers import BasePromptDriver
 from griptape.tokenizers import BaseTokenizer, OpenAiTokenizer
 
 if TYPE_CHECKING:
     from collections.abc import Iterator
 
+    from griptape.common import DeltaMessage, Message, PromptStack
+
 
 @define
 class MockFailingPromptDriver(BasePromptDriver):
```

Comment:
Will need to re-create this for `PromptEngine` instead, since the retry logic is there now.